src/modules/username.rs

```diff
@@ -16,12 +19,25 @@ const USERNAME_ENV_VAR: &str = "USERNAME";
 ///     - The current user isn't the same as the one that is logged in (`$LOGNAME` != `$USER`) [2]
 ///     - The user is currently connected as an SSH session (`$SSH_CONNECTION`) [3]
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    // On Windows the username is mutable when tests are not running.
+    // When proper mocking for elevation is implemented, these
+    // conditions should be removed.
+    #[cfg(all(target_os = "windows", not(test)))]
+    let mut username = context.get_env(USERNAME_ENV_VAR)?;
+
+    #[cfg(any(not(target_os = "windows"), all(target_os = "windows", test)))]
     let username = context.get_env(USERNAME_ENV_VAR)?;
 
     let mut module = context.new_module("username");
     let config: UsernameConfig = UsernameConfig::try_load(module.config);
 
     let is_root = is_root_user();
+
+    #[cfg(all(target_os = "windows", not(test)))]
+    if is_root {
```

Comment:
If you do `cfg!(target_os = "windows")` here (in tests `is_root` is always `false` on Windows), you can simplify this function and remove all the `#[cfg]` statements

Reply:
Sorry, I didn't get you. This part of the statement just changes the username string to "Administrator", no? The `is_root` logic will stay unchanged.

Reply:
`is_root_user()` always returns `false` for tests on Windows, so `not(test)` isn't needed. And, if you used `cfg!` here instead of a cfg-block, `username` could just be `mut` without conditions.